# Aula 04 — Laboratório NumPy e Pandas I

**Disciplina:** Análise e Ciência de Dados
**Curso:** Especialização em Ciência de Dados — IFRN
**Professor:** Plácido Antônio de Souza Neto

---

## Antes de começar

1. Renomeie este arquivo para `aula04_SEUNOME.ipynb`.
2. Mantenha a pasta `dados/` ao lado do notebook, com os arquivos
   `matriculas_ifrn_2026.csv` e `campi_ifrn.xlsx`.
3. Responda **na célula indicada**, sem apagar os enunciados.
4. Execute tudo do zero antes de entregar (`Kernel → Restart & Run All`)
   e confira se não há erros.
5. Entregue o `.ipynb` **já executado** (com as saídas visíveis) no Moodle.

> Os dados são **sintéticos**: nomes de campi e cursos são reais, mas
> matrículas, nomes de estudantes e notas foram gerados aleatoriamente.
> Nenhuma conclusão sobre o IFRN pode ser tirada deles.

## Dicionário de dados — `matriculas_ifrn_2026.csv`

| Coluna | Tipo | Descrição |
|---|---|---|
| `matricula` | texto | Identificador do estudante (8 dígitos, **não é número**) |
| `nome` | texto | Nome fictício do estudante |
| `campus` | texto | Campus do IFRN |
| `curso` | texto | Nome do curso |
| `modalidade` | texto | `Presencial` ou `EaD` |
| `turno` | texto | `Matutino`, `Vespertino` ou `Noturno` |
| `ano_ingresso` | inteiro | Ano de ingresso (2023–2026) |
| `semestre` | inteiro | Semestre de ingresso (1 ou 2) |
| `idade` | inteiro | Idade em anos |
| `nota_final` | decimal | Nota final de 0 a 100 (**tem ausentes**) |
| `frequencia` | decimal | Frequência em % (**tem ausentes**) |
| `situacao` | texto | `Aprovado`, `Reprovado por nota`, `Reprovado por falta`, `Trancado`, `Em curso` |

O arquivo usa **`;` como separador** e **vírgula como separador decimal**.

## Dicionário de dados — `campi_ifrn.xlsx`

Planilha `campi`: `campus`, `cidade`, `regiao`, `ano_criacao`, `num_docentes`
Planilha `cursos`: `curso`, `eixo_tecnologico`, `nivel`, `carga_horaria`


## Questão 1 — Leitura correta do CSV (1,0 ponto)

Leia `dados/matriculas_ifrn_2026.csv` em um DataFrame chamado `df`, de modo que:

- as colunas `nota_final` e `frequencia` fiquem **numéricas** (`float64`);
- a coluna `matricula` fique como **texto**.

Em seguida, mostre o formato (`shape`) do DataFrame.

> **Dica:** os parâmetros `sep`, `decimal` e `dtype` de `pd.read_csv`.

In [1]:
import pandas as pd
import numpy as np

# ESCREVA SEU CÓDIGO AQUI
df = pd.read_csv('dados/matriculas_ifrn_2026.csv',sep=';',decimal = ',',dtype={'nota_final':float,
                                                                 'frequencia':float,
                                                                 'matricula':'str'
                                                                    })
df


,matricula,nome,campus,curso,modalidade,turno,ano_ingresso,semestre,idade,nota_final,frequencia,situacao
0,20261000,Wagner Cavalcante Fernandes,Apodi,Análise e Desenvolvimento de Sistemas,Presencial,Noturno,2024,2,20,34.4,91.4,Reprovado por nota
1,20261001,Zélia Santos Oliveira,Macau,Redes de Computadores,Presencial,Noturno,2025,1,24,70.9,79.3,Aprovado
2,20261002,Zélia Barbosa Costa,Currais Novos,Ciência de Dados,Presencial,Vespertino,2024,1,23,87.8,84.0,Aprovado
3,20261003,Beatriz Medeiros Fernandes,Santa Cruz,Análise e Desenvolvimento de Sistemas,Presencial,Matutino,2024,2,15,49.7,100.0,Reprovado por nota
4,20261004,Ubiratan Bezerra Lima,Currais Novos,Edificações,Presencial,Matutino,2026,2,21,83.5,91.9,Aprovado
...,...,...,...,...,...,...,...,...,...,...,...,...
477,20261477,Emanuel Nogueira Pereira,Natal-Central,Ciência de Dados,Presencial,Matutino,2023,1,27,72.3,88.2,Aprovado
478,20261478,Elaine Galvão Galvão,Pau dos Ferros,Análise e Desenvolvimento de Sistemas,Presencial,Vespertino,2026,2,23,NaN,84.4,Trancado
479,20261479,Daniela Oliveira Rocha,Apodi,Ciência de Dados,EaD,Matutino,2026,1,20,90.6,95.4,Aprovado
480,20261007,Davi Oliveira Galvão,João Câmara,Eletrotécnica,Presencial,Noturno,2026,1,26,54.0,82.7,Reprovado por nota


In [2]:
df['frequencia'].value_counts()

frequencia
100.0    54
96.6      5
72.6      5
78.2      5
80.9      5
         ..
71.1      1
84.5      1
83.2      1
70.0      1
80.5      1
Name: count, Length: 236, dtype: int64

_Sua resposta escrita:_ usar sep com ; para separar as linhas e colunas do dataframe e deixa-lo organizado. colocar o comando decimal=',' para que o pandas reconheça os valores float que contenham virgula, dtype utilizado para converter as colunas nota_final e frequencia para float e matricula para string.




## Questão 2 — Inspeção inicial (1,0 ponto)

Usando `head()`, `info()` e `describe()`, responda **em uma célula de texto**:

a) Quantas linhas e quantas colunas o arquivo tem?
b) Quais colunas possuem valores ausentes e quantos?
c) Qual a nota final **média** e qual a **mediana**? O que a diferença
   entre as duas sugere sobre a distribuição?

In [3]:
# ESCREVA SEU CÓDIGO AQUI
df.head()
df.info()
df.describe()

df.isna().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 482 entries, 0 to 481
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   matricula     482 non-null    object 
 1   nome          482 non-null    object 
 2   campus        482 non-null    object 
 3   curso         482 non-null    object 
 4   modalidade    482 non-null    object 
 5   turno         482 non-null    object 
 6   ano_ingresso  482 non-null    int64  
 7   semestre      482 non-null    int64  
 8   idade         482 non-null    int64  
 9   nota_final    441 non-null    float64
 10  frequencia    473 non-null    float64
 11  situacao      482 non-null    object 
dtypes: float64(2), int64(3), object(7)
memory usage: 45.3+ KB


matricula        0
nome             0
campus           0
curso            0
modalidade       0
turno            0
ano_ingresso     0
semestre         0
idade            0
nota_final      41
frequencia       9
situacao         0
dtype: int64

_Sua resposta escrita:_

a)   482 Linhas e 10 colunas

b)  nota_final possui 41 valores nulos  e frequencia possui 9 valores nulos

c) media nota_final = 69.15  mediana nota_final = 69.70, a diferença entre as duas mostra que uma está próxima da outra em relação ao desvio padrão, media e mediana proximas o desvio é baixo e geram uma normal.


## Questão 3 — Limpeza mínima (1,0 ponto)

O arquivo tem dois defeitos propositais:

1. alguns valores de `campus` vêm com **espaços em branco** nas bordas;
2. existem **linhas duplicadas**.

Corrija os dois problemas e comprove o resultado mostrando, antes e depois,
`df["campus"].nunique()` e `df.duplicated().sum()`.

In [4]:
# ESCREVA SEU CÓDIGO AQUI
df["campus"].value_counts()
df["campus"].nunique()
df.duplicated().sum()

df["campus"] = df["campus"].str.strip() 
df = df.drop_duplicates()

df["campus"].nunique()
df.duplicated().sum()

0

_Sua resposta escrita:_ ao usar o comando value_counts() para verificar a quantidade de valores distintos por nome do campus, percebeu-se que alguns campus se repetiam foram eles: Parnamirim  Natal-Central, Apodi, São Gonçalo do Amarante , Currais Novos, Natal-Zona Norte,Mossoró. o comando nuninque informa todos os valores distintos, como os campus informados possuiam algum caractere a mais o nuninque entende que eles são unicos, então foi necessario aplicar o comando strip() para apagar espaços sobrando e agrupar com os campus ja existentes, após essa mudança ao utilizar o nuninque a quantidade de campus foi para 12. outro problema que havia no dataset eram valores duplicados, ao utilizar o comando duplicated().sum() o pandas identifica quantas linhas duplicadas existem no dataframe para corrigir o problema foi necessário utilizar o comando drop_duplicates() assim removendo os valores duplicados.              



## Questão 4 — `iloc`: seleção por posição (1,0 ponto)

Sem usar `loc`, exiba:

a) as **5 últimas** linhas do DataFrame;
b) as linhas de **posição** 10 a 14 (inclusive), apenas das colunas
   `nome`, `campus` e `nota_final` (use as posições das colunas, não os nomes);
c) o valor da célula na **última linha**, coluna `situacao`.

In [5]:
# ESCREVA SEU CÓDIGO AQUI
#a)
df.iloc[-5:]  # ultimas 5 linhas do dataframe

#b)
df.iloc[10:15, [1,2,9]] # linhas das posicoes 10 a 14 

#c)
df.iloc[-1,11]

'Aprovado'

_Sua resposta escrita:_ 

a) ao utilizar o comando loc[-5:] o pandas busca as 5 ultimas linhas do dataframe.

b) ao utilizar o comando loc[10:15, [1,2,9]] o pandas buscas as linhas da 10 ate 15 e as colunas de indice 1,2,9 que são as colunas nome, campus e nota_final no dataframe.

c) ao utilizar o comando iloc[-1,11] o pandas identifica que -1 é a minha ultima linha do dataframe e ao colocar o 11 após a virgula ele identifica que 11 é o indice da coluna situação no qual temos apenas o resultado da celula especificada. 



## Questão 5 — `loc`: seleção por rótulo (1,0 ponto)

a) Defina `matricula` como índice do DataFrame (em uma cópia chamada `dfm`).
b) Usando `loc`, mostre `nome`, `curso` e `nota_final` da matrícula `20261250`.
c) Explique **em uma frase** por que `dfm.loc["20261250"]` funciona
   mas `dfm.iloc["20261250"]` dá erro.

In [6]:
# ESCREVA SEU CÓDIGO AQUI
# a) 
dfm = df.set_index("matricula").copy().reset_index()
# b)
dfm.loc[dfm["matricula"] =='20261250', ["nome","curso","nota_final"]]



,nome,curso,nota_final
250,Nathan Bezerra Pereira,Química,77.8


_Sua resposta escrita:_

a) para criar um indice com uma coluna do dataframe existente é necessario utilizar o comando set_index(<nomeDaColuna>).reset_index() para resetar o indice e assumir a nova coluna como indice.

b) utilizar o comando loc para realizar o filtro de matricula e após a virgula seleciona as colunas que eu quero exibir na saida do dataframe, nesse caso foram as colunas nome,curso e nota_final

c) loc trabalha com Rótulos o que está escrito no índice e nos nomes das colunas
   iloc trabalha com inteiros de 0 ate n-1, assim como as listas em python, por isso usar dfm.iloc["20261250"] não irá funcionar





## Questão 6 — Filtragem booleana (1,5 ponto)

Crie e mostre o tamanho (`len` ou `.shape`) de cada subconjunto:

a) estudantes do turno **Noturno** com `nota_final` maior ou igual a 70;
b) estudantes da modalidade **EaD** que ingressaram em **2025 ou 2026**;
c) estudantes **aprovados** com `frequencia` abaixo de 80 —
   quantos são? O que esse número diz sobre a regra de aprovação?
d) estudantes cujo `curso` contém a palavra "Sistemas" **ou** "Redes".

In [7]:
# ESCREVA SEU CÓDIGO AQUI
# a) 
df_Noturno = df.loc[(df['turno'] == 'Noturno') & (df['nota_final'] >= 70)]
print(f'estudantes do turno Noturno =  {len(df_Noturno)}')
print(f'Total de linhas e colunas : {df_Noturno.shape}')

# b)
df_ead = df.loc[(df['modalidade'] == 'EaD') & ((df['ano_ingresso'] >= 2025) | (df['ano_ingresso'] <=2026))]
print(f'total de alunos modalidade ead entre 2025 e 2026 : {len(df_ead)}')
print(f'Total de linhas e colunas : {df_ead.shape}')

# c)
df_status = df.loc[(df['situacao'] == 'Aprovado') & (df['frequencia'] < 80)]
print(f'total de alunos aprovados com frequencia menor que 80 : {len(df_status)}')
print(f'Total de linhas e colunas : {df_status.shape}')

# d)

df_curso = df.loc[((df['curso'].str.contains('Sistemas',case=False,na=False)) | (df['curso'].str.contains('Redes',case=False,na=False)))]
print(f'total de alunos dos cursos de Sistemas e Redes : {len(df_curso)}')
print(f'Total de linhas e colunas : {df_curso.shape}')


estudantes do turno Noturno =  81
Total de linhas e colunas : (81, 12)
total de alunos modalidade ead entre 2025 e 2026 : 107
Total de linhas e colunas : (107, 12)
total de alunos aprovados com frequencia menor que 80 : 37
Total de linhas e colunas : (37, 12)
total de alunos dos cursos de Sistemas e Redes : 107
Total de linhas e colunas : (107, 12)


_Sua resposta escrita:_

a) criei um dataframe chamado df_noturno no qual executa os filtros de alunos do turno Noturno e a nota final maior ou igual a 70 dentro da funcao loc, ao aplicar os filtros indicados coloquei dois comandos Print para exibir o total de linhas utilizando a função len() e a função .shape que exibe as linhas e colunas. gerando 81 registros de alunos do turno Noturno que tiveram nota final igual ou maior que 70  

b) criei um dataframe chamado df_ead no qual executa os filtros de alunos da modalidade Ead e ingressaram em 2025 OU 2026 dentro da funcao loc, ao aplicar os filtros indicados coloquei dois comandos Print para exibir o total de linhas utilizando a função len() e a função .shape que exibe as linhas e colunas. gerando 107 registros de alunos da modalidade Ead de 2025 ou 2026 


c) criei um dataframe chamado df_status no qual executa os filtros de situacao do aluno sendo Aprovado e com frequencia abaixo de 80 dentro da funcao loc, ao aplicar os filtros indicados coloquei dois comandos print para exibir o total de linhas utilizando a função len() e a função .shape que exibe as linhas e colunas. gerando 37 registros de alunos aprovados com frequencia menor que 80.  


d) criei um dataframe chamado df_curso no qual executa os filtros de curso do aluno que tenha a palavra Sistemas ou Redes funcao loc, nesse filtro foi interessante pois foi utilizada a função contains() que possue os parametros case e na , case que é tipo o ignore case dependendo do valor booleano se case=True ele vai se importar com as letras maiusculas e minusculas caso o case=False ele simplesmente ignora. e o parametro na caso seja na=True ele vai buscar a palavra que queremos e também trazer valores Nan caso a coluna tenha, caso na=False ele ignora os Nan da coluna e traz apenas o que foi filtrado. Após o uso do filtro o total de linhas retornada foi de 107 registros de alunos de Sistemas ou Redes.
 

## Questão 7 — Valores ausentes (1,5 ponto)

a) Quantos estudantes estão sem `nota_final`? Qual a `situacao` deles?
   (use `value_counts()` sobre o subconjunto filtrado)
b) Preencha os ausentes de `frequencia` com a **mediana** da coluna,
   usando `loc` para atribuir.
c) **Não** preencha `nota_final`. Justifique em uma frase por que
   inventar nota é pior do que inventar frequência aqui.

In [8]:
# ESCREVA SEU CÓDIGO AQUI

## a
df_semNotas = df[df['nota_final'].isna()]
print(df_semNotas['situacao'].value_counts())

## b
df.loc[(df['frequencia'].isna()), 'frequencia'] = df['frequencia'].median()   
df['frequencia'].value_counts()

## c



situacao
Trancado    27
Em curso    14
Name: count, dtype: int64


frequencia
100.0    53
86.1     13
78.2      5
80.9      5
96.6      5
         ..
71.1      1
81.3      1
83.2      1
70.0      1
80.5      1
Name: count, Length: 236, dtype: int64

_Sua resposta escrita:_

a) criei um dataframe chamado df_semNotas para receber todos os alunos que não possuem nota final utilizando a função isna() ela responsavel por pesquisar valores none e NaN, após isso utilizei value_counts() para ver a situação dos Alunos que não tinham nota e foram encontrados 2 situações, sendo elas com 27 alunos com situação "Trancado" e 14 alunos com situação "em curso".

b) apliquei a função fillna para realizar o filtro caso a frequencia seja Nan então os valores Nan recebem a mediana da coluna frequencia. 

c) você não pode aplicar nota final de um aluno baseado no desempenho de outros alunos, podendo gerar a aprovação sem o aluno ter feito nada ou seja desempenho ficticio, aplicar mediana a frequencia também é ruim mas seria um imputação defensavel e deve ser documentada. 





## Questão 8 — Coluna derivada com NumPy (1,0 ponto)

Crie a coluna `faixa` com `np.where`, segundo a regra:

- `nota_final` >= 85 → `"Excelente"`
- 70 <= `nota_final` < 85 → `"Bom"`
- 60 <= `nota_final` < 70 → `"Regular"`
- `nota_final` < 60 → `"Insuficiente"`

**Cuidado:** quem está sem nota deve ficar com `NaN` em `faixa`, e não cair
na última categoria. Mostre `df["faixa"].value_counts(dropna=False)`.

In [ ]:
# ESCREVA SEU CÓDIGO AQUI

df_filtros = df.copy()
condicao = np.where(
                        (df_filtros['nota_final'] >= 85) , "Excelente", 
                            np.where(
                                    (df_filtros['nota_final'] < 85) & (df_filtros['nota_final'] >= 70), "Bom",
                                np.where(
                                       (df_filtros['nota_final'] < 70) & (df_filtros['nota_final'] >= 60), "Regular",
                                    np.where(
                                        (df_filtros['nota_final'] < 60), "Irregular",np.nan
                                        )
                                    )
                                )
                            )
                        
                        
df_filtros['faixa'] = condicao
df_filtros["faixa"] = df_filtros["faixa"].mask(df_filtros["nota_final"].isna())
df_filtros["faixa"].value_counts(dropna=False)


faixa
Bom          175
Regular      120
Irregular    102
Excelente     42
NaN           41
Name: count, dtype: int64

_Sua resposta escrita:_

para criar a coluna chamada faixa utilizando np.where() foi necessario criar um np.where aninhado para colocar todas as condições , inclusive necessario aplicar NaN para valores NaN aplicando a função isna() do pandas, ao final utilizando value_counts() da para verificar os subgrupos de cada faixa por nota, importante ressaltar que foi necessário criar um novo dataframe com a funcao .copy() para evitar o SettingWithCopyWarning indicando a indexação encadeada.



## Questão 9 — Cruzando com o Excel (1,0 ponto)

a) Leia a planilha `campi` de `dados/campi_ifrn.xlsx` em `campi`.
b) Mostre os campi da região **Oeste**.
c) Quantos estudantes do `df` estudam em campi da região Oeste?
   (use `isin` com a lista obtida no item b — **não** digite os nomes à mão)

In [38]:
# ESCREVA SEU CÓDIGO AQUI
 
# a
df_campi = pd.read_excel("dados/campi_ifrn.xlsx",sheet_name='campi')

# b
campis_oeste = df_campi['campus'].loc[(df_campi['regiao'] == 'Oeste')].unique()
print(f"campus da regiao oeste: {campis_oeste}")



# c
total_oeste = len(df.loc[df['campus'].isin(campis_oeste)])
print(f"total de alunos da região oeste: {total_oeste}")


campus da regiao oeste: ['Mossoró' 'Apodi' 'Pau dos Ferros']
total de alunos da região oeste: 137


_Sua resposta escrita:_

a) na leitura do arquivo campi, foi necessaria uma abordagem diferente pois o arquivo está no formato xlsx e foi necessario a utilização da funcao do pandas read_excel e como o pandas ja entende a tabela do excel não é necessário utilizar o comando sep, foi utilizado o comando sheet_name para pegar a aba chamada campi da planilha do excel. 

b) para mostrar os campi da região Oeste, criei uma variavel chamada campis_oeste na qual recebe o filtro utilizando loc para receber apenas os campus da região Oeste, resultando nos Campus Apodi, Mossoró e Pau dos Ferros.

c) aproveitando a variavel campis_oeste apliquei no meu df que possui os dados de matriculas_ifrn, criei um novo dataframe a partir do filtro isin passando como parametro a variavel campis_oeste, resultando nos 137 alunos da região Oeste. 


## Questão 10 — Uma pergunta respondida com dados (1,0 ponto)

Responda, com código **e** com uma frase de interpretação:

> **Os estudantes do turno noturno têm desempenho diferente dos demais?**

Calcule a nota média por turno, a quantidade de estudantes em cada turno e
a taxa de aprovação por turno. Depois escreva, em uma célula de texto, o que
os números mostram — e **uma limitação** dessa conclusão.

> Lembre-se: os dados são sintéticos. Avalia-se o raciocínio, não a conclusão.

In [11]:
# ESCREVA SEU CÓDIGO AQUI
#Noturno       214
#Matutino      150
#Vespertino    116


notas_matutino = df.loc[(df['turno'] == 'Matutino'), 'nota_final'].mean()
notas_verpertino = df.loc[(df['turno'] == 'Vespertino'), 'nota_final'].mean()
notas_noturno = df.loc[(df['turno'] == 'Noturno'), 'nota_final'].mean()

print(f"media de notas turno matutino :{notas_matutino:.2f}")
print(f"media de notas turno vespertino :{notas_verpertino:.2f}")
print(f"media de notas turno noturno :{notas_noturno:.2f}")

total_alunos_m = df.loc[(df['turno'] == 'Matutino')]
total_alunos_v = df.loc[(df['turno'] == 'Vespertino')]
total_alunos_n = df.loc[(df['turno'] == 'Noturno')]
print(f"total de alunos matutino : {len(total_alunos_m)}")
print(f"total de alunos vespertino: {len(total_alunos_v)}")
print(f"total de alunos noturno : {len(total_alunos_n)}")

total_alunos_aprovados_m = df.loc[(df['turno'] == 'Matutino') & (df['situacao'] == 'Aprovado')]
total_alunos_aprovados_v = df.loc[(df['turno'] == 'Vespertino') & (df['situacao'] == 'Aprovado')]
total_alunos_aprovados_n = df.loc[(df['turno'] == 'Noturno') & (df['situacao'] == 'Aprovado')]

print(len(total_alunos_aprovados_m))
print(len(total_alunos_aprovados_v))
print(len(total_alunos_aprovados_n))

print(f"taxa de aprovação dos alunos do matutino {len(total_alunos_aprovados_m)/len(total_alunos_m)*100:.2f}%")
print(f"taxa de aprovação dos alunos do vespertino {len(total_alunos_aprovados_v)/len(total_alunos_v)*100:.2f}%")
print(f"taxa de aprovação dos alunos do noturno {len(total_alunos_aprovados_n)/len(total_alunos_n)*100:.2f}%")



# total de aprovados noturno pela quantidade de alunos no matutino e no vespertino.
#c_m = len(total_alunos_aprovados_n)/len(total_alunos_m)
#print(c_m)
#c_v = len(total_alunos_aprovados_n)/len(total_alunos_v)
#print(c_v)




media de notas turno matutino :70.90
media de notas turno vespertino :71.62
media de notas turno noturno :66.76
total de alunos matutino : 150
total de alunos vespertino: 116
total de alunos noturno : 214
98
65
118
taxa de aprovação dos alunos do matutino 65.33%
taxa de aprovação dos alunos do vespertino 56.03%
taxa de aprovação dos alunos do noturno 55.14%


_Sua resposta escrita:_

é diferente porém seria necessário verificar se a % de aprovação dos alunos do turno Noturno é significantemente proporcional comparado a % de alunos do matutino e vespertino. por exemplo o o turno noturno é o que possui mais alunos, 60 alunos a mais do matutino e 98 a mais do vespertino. sua taxa de aprovação é 55.14% praticamente igual ao vespertino, se realizar um calculo dos 118 alunos aprovados e comparar com o total de alunos do matutino teremos uma taxa de aprovação de 78%, e se comparar com o total de alunos do turno vespertino teremos uma taxa de mais de 100% ou seja tem mais alunos aprovados noturno do que total de alunos no vespertino. 




---

## Entrega

- Arquivo: `aula04_SEUNOME.ipynb`, **executado do início ao fim**.
- Prazo e local: tarefa **"Exercício Prático — Aula 04"** no Moodle.
- Critérios: código correto e legível (60%), uso adequado de `loc`/`iloc`
  e filtragem (20%), interpretação escrita dos resultados (20%).
